core/middleware.py

```diff
@@ -23,3 +48,39 @@ def process_view(self, request, view_func, view_args, view_kwargs):
         if service:
             view_kwargs["service"] = service
         return None
+
+
+class CustomMetricsWithUA(Metrics):
+    """
+    django_prometheus Metrics class but with extra user_agent label for applicable metrics
+    """
+
+    def register_metric(self, metric_cls, name, documentation, labelnames=(), **kwargs):
+        if name in USER_AGENT_METRICS:
+            labelnames = list(labelnames) + ["user_agent"]
+        return super().register_metric(
+            metric_cls, name, documentation, labelnames=labelnames, **kwargs
+        )
```

Comment:
i think your code is right, but it made me notice a couple interesting things so indulge me

compare your approach here to the approach taken in the django-prometheus test app:
```
def register_metric(..., labelnames=(), ...):
    if name in EXTENDED_METRICS:
        labelnames.extend(("view_type", "user_agent_type"))
    return return super().register_metric(...)
```
their approach modifies the passed-in `labelnames` value so whoever calls `register_metric()` would be able check the list they passed in after the function returns and see which labels were added. your approach reassigns `labelnames` to a new list and leaves the passed-in value alone. see what i mean: https://www.online-python.com/lOkDiMp4Kj

looking at how `register_metric()` is used ([example](https://github.com/korfuri/django-prometheus/blob/e96708873c7aaf49f5b571cb08804aaecb45cc9d/django_prometheus/middleware.py#L56)), it doesn't look like anything relies on the passed-in list being updated to contain the new labels. so your approach is fine

the other thing is, it's weird that the default value for `labelnames` is an empty tuple. the passed-in values are all lists so i'd expect the default to be an empty list. also tuples don't have an `.extend()` method so their test app actually requires `labelnames` to be a list :v but it's like that in `django-prometheus` so following suit here is reasonable

Reply:
Interesting, yeah I didn't give the modification vs new list much thought tbh

I'm not actually sure if changing default value of param will mess up the override declaration, but I feel like tuple as default value is just a mistake on their end lol